corehq/form_processor/management/commands/populate_missing_delete_on_values.py

```diff
@@ -1,16 +1,39 @@
+from django.db.models import Q
 from django.core.management.base import BaseCommand
 
 from corehq.form_processor.models import XFormInstance
-from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+from corehq.sql_db.util import paginate_query_across_partitioned_databases
+
+from dimagi.utils.chunked import chunked
 
 
 class Command(BaseCommand):
-    help = "One time use. Populate missing XFormInstance.deleted_on values"
+    help = "One time use to populate missing XFormInstance.deleted_on values"
 
     def handle(self, **options):
-        for db in get_db_aliases_for_partitioned_query():
-            queryset = XFormInstance.objects.using(db).filter(deleted_on__isnull=True)
-            for xform in queryset:
-                if xform.doc_type == 'XFormInstance-Deleted':
-                    xform.deleted_on = xform.server_modified_on
-                    xform.save()
+        chunk_size = 100
+        old_deleted_state = 64
+
+        def iter_all_ids():
+            rows = paginate_query_across_partitioned_databases(
+                XFormInstance,
+                Q(deleted_on__isnull=True),
```

Comment:
Is it possible to include the _deleted_ state filter here? (`state & 64 = 64`) That would greatly reduce the number of records returned by this iteration.

Reply:
added the extra filter!